contrib/endpoints/src/api_manager/context/service_context.cc

```diff
@@ -24,57 +23,25 @@ namespace context {
 
 namespace {
 
-// Default Cloud Trace URL. Points to prod Cloud Trace.
-const char kCloudTraceUrl[] = "https://cloudtrace.googleapis.com";
-
-// Default maximum time to aggregate traces.
-const int kDefaultAggregateTimeMillisec = 1000;
-
-// Default maximum amount of traces to aggregate. The amount should ensure
-// the http request payload with the aggregated traces not reaching MB in size.
-const int kDefaultTraceCacheMaxSize = 100;
-
-// Default trace sample rate, in QPS.
-const double kDefaultTraceSampleQps = 0.1;
-
-// The time window to send intermediate report for Grpc streaming (second).
-// Default to 10s.
-const int kIntermediateReportInterval = 10;
-
 const char kHTTPHeadMethod[] = "HEAD";
 const char kHTTPGetMethod[] = "GET";
-const char kFirebaseAudience[] =
-    "https://staging-firebaserules.sandbox.googleapis.com/"
-    "google.firebase.rules.v1.FirebaseRulesService";
 }
 
-ServiceContext::ServiceContext(std::unique_ptr<ApiManagerEnvInterface> env,
+ServiceContext::ServiceContext(std::shared_ptr<GlobalContext> global_context,
                                std::unique_ptr<Config> config)
-    : env_(std::move(env)),
+    : global_context_(global_context),
       config_(std::move(config)),
-      service_account_token_(env_.get()),
-      service_control_(CreateInterface()),
-      cloud_trace_aggregator_(CreateCloudTraceAggregator()),
-      is_auth_force_disabled_(
-          config_->server_config() &&
-          config_->server_config()->has_api_authentication_config() &&
-          config_->server_config()
-              ->api_authentication_config()
-              .force_disable()) {
-  intermediate_report_interval_ = kIntermediateReportInterval;
-
-  // Check server_config override.
-  if (config_->server_config() &&
-      config_->server_config()->has_service_control_config() &&
-      config_->server_config()
-          ->service_control_config()
-          .intermediate_report_min_interval()) {
-    intermediate_report_interval_ = config_->server_config()
-                                        ->service_control_config()
-                                        .intermediate_report_min_interval();
-  }
+      service_control_(CreateInterface()) {
+  config_->set_server_config(global_context_->server_config());
```

Comment:
Why do you need this method? The global context should be available when config is built right? The reason I ask is the number of places we need to maintain the server_config. It is available in global_context, so why not have config_ object built with global_context since that is where the env_ object is also present and we don't have to keep passing the configs around. That way, if server_config is updated, it is updated in the global context and everyone who know the global context will see the new server config.